php/libraries/Recording.class.inc

```diff
@@ -0,0 +1,425 @@
+                       ON (f.SessionID = s.ID)
+                     LEFT JOIN candidate c
+                       ON (s.CandID = c.CandID)
+                     LEFT JOIN  physiological_output_type pot
+                       ON (pot.PhysiologicalOutputTypeID = f.PhysiologicalOutputTypeID)
+                     LEFT JOIN physiological_modality pm
+                       ON (pm.PhysiologicalModalityID = f.PhysiologicalModalityID)
+                     LEFT JOIN (
+                       SELECT PhysiologicalFileID, FilePath
+                       FROM physiological_channel
+                       LIMIT 1
+                     ) pc
+                       ON (pc.PhysiologicalFileID = f.PhysiologicalFileID)
+                     LEFT JOIN (
+                       SELECT PhysiologicalFileID, FilePath
+                       FROM physiological_electrode
+                       LIMIT 1
+                     ) pe
+                       ON (pe.PhysiologicalFileID = f.PhysiologicalFileID)
+                     LEFT JOIN (
+                       SELECT PhysiologicalFileID, FilePath
+                       FROM physiological_task_event
+                       LIMIT 1
+                     ) pte
+                       ON (pte.PhysiologicalFileID = f.PhysiologicalFileID)
+                     LEFT JOIN physiological_archive pa
+                       ON (pa.PhysiologicalFileID = f.PhysiologicalFileID)
+                     WHERE f.PhysiologicalFileID = :v_fileid
+                    ',
+                    array('v_fileid' => $fileid)
+                );
+            if (empty($dbrow)) {
+                throw new \NotFound('There is no file with that FileID');
+            }
+            $this->_fileid       = $fileid;
+            $this->_filename     = basename($dbrow['filelocation']);
+            $this->_filelocation = $dbrow['filelocation'];
+            $this->_outputtype   = $dbrow['outputtype'];
+            $this->_acquisitionmodality = $dbrow['acquisitionmodality'];
+            $this->_filetype            = $dbrow['filetype'];
+            $this->_centerid            = (int) $dbrow['centerid'];
+            $this->_entitytype          = $dbrow['entitytype'];
+            $this->_channel_file_path   = $dbrow['channel_file_path'];
+            $this->_electrode_file_path = $dbrow['electrode_file_path'];
+            $this->_event_file_path     = $dbrow['event_file_path'];
+            $this->_archive_file_path   = $dbrow['archive_file_path'];
+        }
+    }
+
+    /**
+     * Get a specific file header from physiological_parameter file
+     *
+     * @param string $headername The parameter name to retrieve
+     *
+     * @return string The value or an empty string
+     */
+    public function getMetadata(string $headername): string
+    {
+        $dbrow = \NDB_Factory::singleton()
+            ->database()
+            ->pselectRow(
+                '
+                       SELECT
+                         Value
```

Comment:
If you select only parameter from 1 table, it is generally more efficient to use sub-queries than JOIN.

Reply:
what do you mean?

Reply:
Instead of creating a temporary table (JOIN) incorporating many tables , it is quite often more efficient in resources to put subquery in the WHERE clause.

```
SELECT value
FORM physiological_parameter_file
WHERE physiological_parameter_file = :v_fileid
 AND ParameterTypeID = (
    SELECT Name 
    FROM parameter_type
    WHERE Name = :v_headername
)
```
Since `PhysiologicalFileID` is in the WHERE clause and no additional info is required from `physiological_file`, it is not needed in this query. Unless you used it as a way to return no data if not present, than there is a better way than join (with another nested SELECT.

Reply:
ok. Thanks! Done in the next commit